Dockerfile

```diff
@@ -0,0 +1,157 @@
+FROM ubuntu:16.04 as portal-modules
+
+# Set timezone to UTC by default
+RUN ln -sf /usr/share/zoneinfo/Etc/UTC /etc/localtime
+
+# Use unicode
+RUN locale-gen C.UTF-8 || true
+ENV LANG=C.UTF-8
+
+RUN apt-get update \
+  && mkdir -p /usr/share/man/man1 \
+  && apt-get install -y \
+    git mercurial xvfb \
+    locales sudo openssh-client ca-certificates tar gzip parallel \
+    net-tools netcat unzip zip bzip2 gnupg curl wget build-essential curl wget
+
+# nodejs
+RUN apt autoremove -y nodejs
+RUN curl -sL https://deb.nodesource.com/setup_9.x | bash -
+RUN apt install nodejs -y
+
+# paratii-portal
+WORKDIR /paratii-portal
+ADD package.json /paratii-portal/package.json
+RUN npm install
+
+FROM ubuntu:16.04
+
+WORKDIR /build
+
+# Set timezone to UTC by default
+RUN ln -sf /usr/share/zoneinfo/Etc/UTC /etc/localtime
+
+# Use unicode
+RUN locale-gen C.UTF-8 || true
+ENV LANG=C.UTF-8
+
+RUN apt-get update \
+  && mkdir -p /usr/share/man/man1 \
+  && apt-get install -y \
+    git mercurial xvfb \
+    locales sudo openssh-client ca-certificates tar gzip parallel \
+    net-tools netcat unzip zip bzip2 gnupg curl wget
+
+RUN apt-get update
+RUN apt install build-essential chrpath libssl-dev libxft-dev -y && apt install libfreetype6 libfreetype6-dev apt-transport-https git -y
+RUN apt install libfontconfig1 libfontconfig1-dev -y
+
+# openssl 1.1.0
+
+RUN wget https://www.openssl.org/source/openssl-1.1.0.tar.gz \
+ && tar xvzf openssl-1.1.0.tar.gz \
+ && cd openssl-1.1.0 \
+ && ./config -Wl,--enable-new-dtags,-rpath,'$(LIBRPATH)' \
+ && make \
+ && make install \
+ && ln -s /usr/local/lib/libcrypto.so.1.1 /usr/lib/ \
+ && rm /usr/bin/openssl \
+ && ln -s /usr/local/ssl/bin/openssl /usr/bin/openssl
+
+# parity 1.9.0
+RUN wget -q https://parity-downloads-mirror.parity.io/v1.9.0/x86_64-unknown-linux-gnu/parity_1.9.0_amd64.deb && dpkg -i parity_1.9.0_amd64.deb
+
+# nodejs
+RUN apt autoremove -y nodejs
+RUN curl -sL https://deb.nodesource.com/setup_9.x | bash -
+RUN apt install nodejs -y
+
+# yarn
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+RUN apt update
+RUN apt install yarn
+
+# install jq
+RUN JQ_URL="https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/jq-latest" \
+  && curl --silent --show-error --location --fail --retry 3 --output /usr/bin/jq $JQ_URL \
+  && chmod +x /usr/bin/jq \
+  && jq --version
+
+# install java 8
+#
```

Comment:
What do we need Java for?

Reply:
(in general, it seems that we have  lots of software installed here that we do not need, can you try to clean it up a bit?)

Reply:
yes @jellegerbrandy you have right! I have some changes locally that I need to push.